### DISCUSSION TITLE
Biff's Studio + Various Belper Pubs (4h) 29/6/23

### DISCUSSION THREAD

**Discussion by richardbirkin** [SuperHyperInstantFutureTime/TrackShift#80] [Meetings]
We got the funding and the go ahead from Innovate UK!

We met to discuss our work strategy for the next 6 months as we work on the IUK project - TrackShift for Record Labels (alpha)

Also to go through ideas and work on SLC's (deliverables)

I had already made some notes about what the user journey should look like, based on the interviews with our first batch of labels that we're researching with. 

### Labels with Existing Sales

```mermaid
flowchart TD
A[upload a statement] --> B[list of products and total sales for each]
B --> C[add costs per product]
B --> D[add share splits and contacts]
C --> D
C --> E[dashboard]
D --> E
E --> F[create artist reports]
F --> G[send]
```

The idea being that the only thing the user has to enter is the costs (outgoings) and share splits. The rest is auto populated from the contents of an uploaded CSV file. 

It's also important to think about the above in the context of a label just starting out, or creating a new product in TrackShift before it has shown up in Sales Reports. 

### Labels starting out

```mermaid
flowchart TD
A[create a product] --> B[add costs]
B --> C[add share splits and contact details]
C --> D[await sales data / connect bandcamp / notify the parties]
```

From this we created the first **SLC**

Note: A singular **SLC** is a deliverable set of features that is Simple, Loveable and Complete. 

Simple - connect a TS Label User to Bandcamp and/or use DSP APIs to grab meta data and assets e.g. artwork
Loveable - user interface (a loading animation featuring the record sleeves?), seeing a table populated with your album covers without lifting a finger
Complete - TS has all available info on each release and/or is conntected to bandcamp so the user never has to upload a sales files from there again. 

So, the first flowchart in terms of functionality would look like this:

```mermaid
flowchart TD
A[upload a statement] --> B[retrieve meta and assets from Spotify/AppleMusic using EANs and ISRCs]
B --> C[populate table with releases, sales summary for each, artwork]
C --> D[add share splits and contacts]
C --> E[dashboard]
D --> E
E --> F[create artist reports]
F --> G[send]
```

## Objects

We made a basic list of objects that will exist in TrackShift:

- TS User
- Album
- Product
- Artist
- Tune
- Upload
- Statement/Report
- Payee
- Payout
- Metadata
- Outgoing (Cost)
- Usage (Sale or Stream)

The purpose here is to create a UML / Heirarchy 

## Homepage

We're still keen on the idea that when a user lands, they can instantly start using TrackShift. So we're going to have an upload box as the hero, with all the marketing content below the fold. 

Explain simplicity, security. "The path of least resistance"

Login should use Authwave. Email is the source of truth. 

## Notifications (SLC 2)

A new feature that Greg came up with is the notification bell being the place where we talk to the user, instead of popups and all the stuff that makes the web annoying. 

There will always be something for new users. 

Maybe the user can also respond here. Almost like a chat. Thumbs up/down. Conversational. 
Release notes for the user to read about. 
Blog posts. 

This will be **SLC 2**

## Code Releases

Going forward, we'll use release numbers every time we commit new code to the public version. 

## Data Overwrite / Uploads (SLC 3)

We talked about something that we'll need to constantly test - if a user uploads duplicate data, TrackShift should know that it's seen that row of data before, and only update the existing data NOT create new data. 

This is the same for when a user is connected via the Bandcamp API. Say they upload a statement as well (or by mistake). TS should know the data. Timestamp? Sales ID? There will be something to identify each row as unique and referenceable.

This will be SLC 3: A user should be able to see the things they have uploaded.  Filenames, summary of contents, download original file, delete.

## Other Business

I was approached on LinkedIn by an old friend who wants to donate some time in exchange for the experience of doing product management for a startup. Greg is keen, so am I. We'll hook that up then. 

Also Robot Mascot sent their invoice for the pitch development work that we've got them down for in the budget. Need to shift some numbers around in the budget as the cost is slightly higher than estimated, but we've secured two maybe three labels without having to travel so have saved money there.

**Comment by g105b** [comment]
Great stuff. Did you hear back from Bandcamp, or are they still on holiday?

**Comment by richardbirkin** [comment]
I documented the responses on #79
